Realm/Realm/Sync/Session.cs

```diff
@@ -185,5 +225,34 @@ internal void CloseHandle(bool waitForShutdown = false)
         }
 
         internal void ReportErrorForTesting(int errorCode, string errorMessage, bool isFatal) => Handle.ReportErrorForTesting(errorCode, errorMessage, isFatal);
+
+        internal static void RaiseError(Session session, Exception error)
+        {
+            var args = new ErrorEventArgs(error);
+            Error?.Invoke(session, args);
+        }
+
+        private void SubscribeNotifications()
+        {
+            var cb = new Action<string>((propertyName) => NotifyPropertyChanged(propertyName));
```

Comment:
My intention was to store it in the `_propertyChangedNotificationTokens`, which is a [NotificationToken])https://github.com/realm/realm-dotnet/blob/1152c35aabef4252232a2db49aca16d3bf6ff2f1/Realm/Realm/DatabaseTypes/INotifiable.cs#L40). Unfortunately, I sent the tokens instead of the callback. Would you agree that fixing this would be enough?

Reply:
You could pass `cb` as the first argument to `NotificationToken.Create`, yeah, and then capture `nativeTokens` in the unsubscribe lambda. It would look a little bit weird, I think, but it will work.

Personally I'm not sure I'd use the `NotificationToken` machinery here in the first place, because we're only handling subscribing and unsubscribing for the callbacks internally and there isn't an API that looks like `RealmCollectionBase<T>.SubscribeForNotifications`, but I'm not opposed to it if you prefer it for the sake of consistency.

Reply:
@nirinchev What's your opinion on this?

Reply:
As mentioned in my review, I think that creating a weak handle around the session will simplify the lifetime tracking and most things should work out of the box and will also be more consistent with the design of notifications in `RealmObjectBase`.